src/components/navbar/navbar.component.tsx

```diff
@@ -55,6 +57,16 @@ const Navbar: React.FC<NavbarProps> = ({
     }
   };
 
+  const [switcherToggler, setSwitcherToggler] = React.useState(false);
+
+  const handleSwitcherClick = () => {
```

Comment:
Wrap the CB in `useCallback` and make it a toggler (`setSwitcherToggle(state => !state)`)